### PR TITLE
firmware: update esp32c3-hal

### PR DESCRIPTION
**Pull request by ImUrX**

### CODE REVIEW

File: firmware/Cargo.toml [hunk 1]
```diff
@@ -132,7 +132,7 @@ smoltcp = { version = "0.8", default-features = false, features = [
 
 # Platform independent traits
 embedded-hal = "0.2"
-embedded-svc = { version = "0.23", default-features = false, optional = true, features = [
+embedded-svc = { version = "0.23.1", default-features = false, optional = true, features = [
```
Comment (TheButlah) [inline firmware/Cargo.toml]: any particular reason we used .1?

Comment (ImUrX) [reply firmware/Cargo.toml]: its the one that esp-wifi is using, we should really look into what dependencies esp-wifi has when we use it :sweat:

Comment (TheButlah) [reply firmware/Cargo.toml]: In that case just leave it as `0.23` - cargo will resolve the two.

Comment (ImUrX) [reply firmware/Cargo.toml]: didnt happen when having different ``-rt`` versions :rofl:, i wont trust cargo in things that affect linking.

**Review by TheButlah — APPROVED** [review]

**Pull request merged**